index.ts

```diff
@@ -0,0 +1,63 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import depd from 'depd';
+
+const deprecate = depd('hsts');
+
+const DEFAULT_MAX_AGE = 180 * 24 * 60 * 60;
+
+interface HstsOptions {
+  includeSubDomains?: boolean;
+  maxAge?: number;
+  preload?: boolean;
+  setIf?: (req: IncomingMessage, res: ServerResponse) => boolean;
+}
+
+function alwaysTrue () {
+  return true;
+}
+
+export = function hsts (options: HstsOptions = {}) {
+  if ('includeSubdomains' in options) {
+    deprecate('The "includeSubdomains" parameter is deprecated. Use "includeSubDomains" (with a capital D) instead.');
+  }
+
+  if ('setIf' in options) {
+    deprecate('The "setIf" parameter is deprecated. Refer to the documentation to see how to set the header conditionally.');
+  }
+
+  const { maxAge = DEFAULT_MAX_AGE, setIf = alwaysTrue } = options;
```

Comment:
The original behavior isn't ideal, but this is actually a breaking change. The old line looked like this:

```js
var maxAge = options.maxAge != null ? options.maxAge : DEFAULT_MAX_AGE
```

That means that the a parameter of `{}` and `{maxAge: null}` were equivalent. Now, they're not. I filed #26 to fix it when we next make a breaking change, but we should fix the behavior here.